lib/ansible/modules/remote_management/manageiq/manageiq_config.py

```diff
@@ -0,0 +1,215 @@
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: manageiq_config
+
+short_description: Module for managing ManageIQ configuration.
+version_added: '2.5'
+author: Ian Tewksbury (@itewk)
+description:
+    - "Module for managing ManageIQ or CloudForms Managment Engine (CFME) configuration via the local rails runner on the Appliance."
+
+options:
+    name:
+        description:
+            - Name of the ManageIQ config element to modify.
+        required: True
+    value:
+        description: Dictionary to set as the value for the ManageIQ config option. Any values not given will be left at current value.
+        required: False
+        default: {}
+
+'''
+
+EXAMPLES = '''
+# set the smtp settings
+- manageiq_config:
+    name: smtp
+    value:
+      from: cfme@example.com
+      host: postfix.example.com
+      port: 25
+      domain: example.com
+
+# set generic worker memory threshold
+- manageiq_config:
+    name: workers
+    value:
+      worker_base:
+        queue_worker_base:
+          generic_worker:
+            memory_threshold: '900.megabytes'
+'''
+
+RETURN = '''
+name:
+    description: Name of the ManageIQ config option that is being modified.
+    type: str
+    returned: always
+original_value:
+    description: The origional value of the ManageIQ config option being modified before modification.
+    type: dict
+    returned: always
+value:
+    description: The value of the ManageIQ config option being modified after modification.
+    type: dict
+    returned: always
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.basic import json
+import time
+import collections
+import copy
+
+VMDB_PATH = '/var/www/miq/vmdb'
+CONFIRM_UPDATE_MAX_RETRY = 10
+CONFIRM_UPDATE_SLEEP_INTERVRAL = 1
+
+
+# @source https://gist.github.com/angstwad/bf22d1822c38a92ec0a9
+def dict_merge(dct, merge_dct):
+    """ Recursive dict merge. Inspired by :meth:``dict.update()``, instead of
+    updating only top-level keys, dict_merge recurses down into dicts nested
+    to an arbitrary depth, updating keys. The ``merge_dct`` is merged into
+    ``dct``.
+    :param dct: dict onto which the merge is executed
+    :param merge_dct: dct merged into dct
+    :return: None
+    """
+    for k, v in merge_dct.items():
+        if (k in dct and isinstance(dct[k], dict)
+                and isinstance(merge_dct[k], collections.Mapping)):
+            dict_merge(dct[k], merge_dct[k])
+        else:
+            dct[k] = merge_dct[k]
+
+
+def get_manageiq_config_value(module, name):
+    """ Gets the current value for the given config option.
+    :param module: AnsibleModule making the call
+    :param name: ManageIQ config option name
+    :return: dict of value of the ManageIQ config option
+    """
+    (get_value_rc, get_value_out, get_value_err) = module.run_command([
```

Comment:
It'll look more clean if you name these `returncode`, `out`, and `_` (the underscore denotes you're not going to use that, and it's a common pattern in Python). You can also drop the parenthesis, they're not needed

Reply:
okay, though using `get_value_err` in the error case so will change that to `err` instead of `-`